### PR TITLE
AzureStack Storage Powershell Changes for GA

### PR DESCRIPTION
**Pull request by cheguv**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -2,6 +2,9 @@
 <Include xmlns="http://schemas.microsoft.com/wix/2006/wi">
     <Fragment>
         <DirectoryRef Id="PowerShellFolder">
+            <Component Id="cmpDC507D3B034FD0F9887EC67302B6F5A0" Guid="*">
+                <File Id="filBF2E03D229851ADE6748EBFCE4288A27" KeyPath="yes" Source="$(var.sourceDir)\PowerShellSetup.Test.dll" />
+            </Component>
```
Comment (bganapa) [inline setup/azurecmdfiles.wxi]: This is not needed i think, mark pls confirm

File: src/Common/Commands.Common.Storage/AzureStorageContext.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ public AzureStorageContext(CloudStorageAccount account)
         /// Proivides a private constructor for building empty instance which
         /// contains no account information.
         /// </summary>
-        private AzureStorageContext()
+        protected AzureStorageContext()
```
Comment (bganapa) [inline src/Common/Commands.Common.Storage/AzureStorageContext.cs]: we should make sure that we are updating the AzureRM.Storage module version

File: src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1 [hunk 3]
```diff
@@ -74,7 +74,36 @@ CmdletsToExport = '*'
 VariablesToExport = '*'  
   
 # Aliases to export from this module  
-AliasesToExport = @()
+AliasesToExport = @(
```
Comment (bganapa) [inline src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1]: Is this needed? Shouldn't this happen automatically, if our intent is to expose all aliases

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj [hunk 4]
```diff
@@ -54,10 +54,17 @@
...
+    <Reference Include="Microsoft.Azure.Gallery">
+      <HintPath>..\..\..\Packages\Microsoft.Azure.Gallery.2.6.2-preview\lib\net40\Microsoft.Azure.Gallery.dll</HintPath>
+    </Reference>
     <Reference Include="Microsoft.Azure.KeyVault.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.KeyVault.Core.1.0.0\lib\net40\Microsoft.Azure.KeyVault.Core.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Management.Authorization, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\Packages\Microsoft.Azure.Management.Authorization.2.0.0\lib\net40\Microsoft.Azure.Management.Authorization.dll</HintPath>
```
Comment (bganapa) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj]: Authorization is 2.0.1 now

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj [hunk 5]
```diff
@@ -70,9 +77,9 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.0.6068.31305-prerelease\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.AzureStack.AzureConsistentStorage, Version=0.10.6.0, Culture=neutral, processorArchitecture=MSIL">
+    <Reference Include="Microsoft.AzureStack.AzureConsistentStorage, Version=0.10.7.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (bganapa) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj]: i thought you published 0.10.8

File: tools/LocalFeed/Microsoft.AzureStack.AzureConsistentStorage.0.10.7/lib/net40/Microsoft.AzureStack.AzureConsistentStorage.xml [hunk 6]
```diff
@@ -0,0 +1,5685 @@
+<?xml version="1.0"?>
```
Comment (bganapa) [inline tools/LocalFeed/Microsoft.AzureStack.AzureConsistentStorage.0.10.7/lib/net40/Microsoft.AzureStack.AzureConsistentStorage.xml]: can you remove the local feed and publish 0.10.8-preview nuget package

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj (hunk 4 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj]: fixed.

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj (hunk 5 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj]: fixed

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj [hunk 7]
```diff
@@ -70,9 +77,9 @@
...
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.AzureStack.AzureConsistentStorage.0.10.6\lib\net45\Microsoft.AzureStack.AzureConsistentStorage.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.AzureStack.AzureConsistentStorage.0.10.7\lib\net45\Microsoft.AzureStack.AzureConsistentStorage.dll</HintPath>
```
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/Commands.AzureStackStorage.Test.csproj]: fixed now using Version=0.10.8.0-preview
updated packages.config as well

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/packages.config [hunk 8]
```diff
@@ -7,7 +7,7 @@
   <package id="Microsoft.Azure.Management.Resources" version="2.18.14-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.Framework" version="1.0.6068.31305-prerelease" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.HttpRecorder" version="1.0.6068.31305-prerelease" targetFramework="net45" />
-  <package id="Microsoft.AzureStack.AzureConsistentStorage" version="0.10.6" targetFramework="net45" />
+  <package id="Microsoft.AzureStack.AzureConsistentStorage" version="0.10.7" targetFramework="net45" />
```
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage.Tests/packages.config]: fixed
package id="Microsoft.AzureStack.AzureConsistentStorage" version="0.10.8-preview" targetFramework="net45"

File: setup/azurecmdfiles.wxi (hunk 1 above)
Comment (cheguv) [inline setup/azurecmdfiles.wxi]: removed and was able to build.

File: src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1 (hunk 3 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1]: When i didnt include these, aliases were not working, after this it started working.

File: src/Common/Commands.Common.Storage/AzureStorageContext.cs (hunk 2 above)
Comment (cheguv) [inline src/Common/Commands.Common.Storage/AzureStorageContext.cs]: Awaiting Mark's update

File: src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1 (hunk 3 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1]: Awaiting Mark's update

File: setup/azurecmdfiles.wxi [hunk 9]
```diff
@@ -364,9 +364,18 @@
                         </Component>
                     </Directory>
                     <Directory Id="dir8A823C7DB199DCB444D0A6388BC3C769" Name="AzureRM.AzureStackStorage">
+                        <Component Id="cmpC180F2B7917E4BFE6CA02D4CE6A32058" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: why are these files added in the installer?  You should not require any resource manager cmdlets modules as part of this module

File: setup/azurecmdfiles.wxi [hunk 10]
```diff
@@ -433,6 +475,9 @@
                         <Component Id="cmpC1B4A054FDF189529C7F141DFF34F627" Guid="*">
                             <File Id="fil5CE63C059CEB87D033F4CB5F72E368D8" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Newtonsoft.Json.dll" />
                         </Component>
+                        <Component Id="cmp65415E5150B588AE9E3C610865988601" Guid="*">
+                            <File Id="fil0233B880D82CAB2CE92A78465E7A3517" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\ResourceManagerStartup.ps1" />
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: Not needed as well

File: setup/azurecmdfiles.wxi [hunk 11]
```diff
@@ -376,18 +385,51 @@
...
+                        <Component Id="cmpA45ABC0D8FA50C56AF89BE99433CD421" Guid="*">
+                            <File Id="fil466357D7D1CF19EC2CDD3B512283566D" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Commands.ResourceManager.Cmdlets.dll" />
+                        </Component>
+                        <Component Id="cmp5B273CF5F9DA506A81761968AFB49352" Guid="*">
+                            <File Id="fil010B3D9994159D5E519D40FFD2666851" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Commands.ResourceManager.Cmdlets.dll-Help.xml" />
+                        </Component>
                         <Component Id="cmpF59DACE5A099A1224C2EBE519139EA1B" Guid="*">
                             <File Id="fil1E856425087E9F8628785419EBD1E1A4" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Commands.ResourceManager.Common.dll" />
                         </Component>
+                        <Component Id="cmp063FB4993B50731237083D3B8F62D7F5" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: None of these should be required.

File: setup/azurecmdfiles.wxi [hunk 12]
```diff
@@ -376,18 +385,51 @@
...
+                        <Component Id="cmp34FA8B1CC091698D28E81F0BFF60B5A0" Guid="*">
+                            <File Id="filABDB4089DB5EAB2F1CFD75BD586AA10F" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Commands.Tags.format.ps1xml" />
+                        </Component>
                         <Component Id="cmp5C982BDD1D0615791970EB8BF2D2E9AF" Guid="*">
                             <File Id="fil1E78C53F64DEF5DB27FFEA88E0DE493A" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Common.dll" />
                         </Component>
                         <Component Id="cmp641D1C76590267170546F7C8FE37B488" Guid="*">
                             <File Id="fil4C8CFBADCB25D9A9493139419D43A175" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Common.NetFramework.dll" />
                         </Component>
+                        <Component Id="cmpA0061A6709DC48860055AC65A45FA4C6" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: Neither of these should be required

File: setup/azurecmdfiles.wxi [hunk 13]
```diff
@@ -364,9 +364,18 @@
                         </Component>
                     </Directory>
                     <Directory Id="dir8A823C7DB199DCB444D0A6388BC3C769" Name="AzureRM.AzureStackStorage">
+                        <Component Id="cmpC180F2B7917E4BFE6CA02D4CE6A32058" Guid="*">
+                            <File Id="fil5DDE3EBAA318EB86C9B3B6C5CF3D1EFC" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\AzureResourceManager.psd1" />
+                        </Component>
                         <Component Id="cmp613FC7082DD4AE634B8DADCFF3516FB5" Guid="*">
                             <File Id="filC9E1E04075F62F0EC70A11C68DD4C754" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\AzureRM.AzureStackStorage.psd1" />
                         </Component>
+                        <Component Id="cmp312E0393299B5A07249BF97D92D2360D" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: These should not be added

File: setup/azurecmdfiles.wxi [hunk 14]
```diff
@@ -376,18 +385,51 @@
                         <Component Id="cmp9ED8706300C8CFD17C8CB62730F15BDA" Guid="*">
                             <File Id="fil5288D78B57719054D2775BC89792069F" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.AzureStackStorage\Microsoft.Azure.Commands.Common.Authentication.dll" />
                         </Component>
+                        <Component Id="cmpA45ABC0D8FA50C56AF89BE99433CD421" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: These should not be added

File: src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1 (hunk 3 above)
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1]: Cmdlets are automatically added, aliases are not.  Add any new aliases here

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Commands.AzureStackStorage.csproj [hunk 15]
```diff
@@ -214,6 +212,10 @@
       <Project>{3819d8a7-c62c-4c47-8ddd-0332d9ce1252}</Project>
       <Name>Commands.ResourceManager.Common</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\Resources\Commands.Resources\Commands.Resources.csproj">
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Commands.AzureStackStorage.csproj]: You should not add this reference.  This brings over all of the additional unneeded files you have in the .wxi file.  It also puts a tight coupling between these two modules.  Any needed methods from resource manager should go into common code, or directly into this module

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Nouns.cs [hunk 16]
```diff
@@ -16,52 +16,22 @@ namespace Microsoft.AzureStack.AzureConsistentStorage.Commands
 {
     class Nouns
     {
-        public const string Prefix = "ACS";
+        public const string Prefix = "AzS";
```
Comment (markcowl) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Nouns.cs]: Why AzS?  This is going to look weird.  Why not Azs?

Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Nouns.cs]: Checking with the PM team's if they are fine with the name "Azs"

File: src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1 (hunk 3 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/AzureRM.AzureStackStorage.psd1]: All the added aliases are new aliases.

File: src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Nouns.cs (hunk 16 above)
Comment (cheguv) [inline src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Nouns.cs]: PM's are fine with Azs, new change pushed.

**Comment by markcowl** [comment]
@cheguv @bganapa Per discussion, either remove the reference to resource manager cmdlets, or edit ```tools\Installer\generate.ps1``` to remove the unneeded files in the wxi (only the dlls should be needed if you keep the reference)

**Comment by bganapa** [comment]
powershell demand successful - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1674/

File: src/Common/Commands.Common.Storage/AzureStorageContext.cs (hunk 2 above)
Comment (bganapa) [inline src/Common/Commands.Common.Storage/AzureStorageContext.cs]: I updated the version of the module with another PR

**Comment by cheguv** [comment]
Hi Mark, all fixes have been made as per our discussion.
